src/Orchard.Data/ServiceCollectionExtensions.cs

```diff
@@ -25,6 +27,8 @@ public static IServiceCollection AddDataAccess(this IServiceCollection services)
             services.TryAddDataProvider(name: "Sql Server", value: "SqlConnection", hasConnectionString: true);
             services.TryAddDataProvider(name: "Sql Lite", value: "Sqlite", hasConnectionString: false);
 
+            SqlDialectFactory.RegisterSqlDialect("diagnosticdbconnection", new SqlServerDialect());
```

Comment:
@sebastienros The Dialect stuff in YesSql stops me from wrapping up DBConnections unless I register them. Any ideas?